spec/unleash/constraint_spec.rb

```diff
@@ -72,5 +72,306 @@
+      expect(constraint.matches_context?(context)).to be_truthy
+
+      constraint = Unleash::Constraint.new('env', 'STR_CONTAINS', ['dev'])
+      expect(constraint.matches_context?(context)).to be_truthy
+
+      constraint = Unleash::Constraint.new('env', 'STR_CONTAINS', ['development'])
+      expect(constraint.matches_context?(context)).to be_truthy
+
+      constraint = Unleash::Constraint.new('env', 'STR_CONTAINS', ['DEVELOPMENT'])
+      expect(constraint.matches_context?(context)).to be_falsey
+    end
+
+    it 'matches based on property NUM_EQ value' do
+      context_params = {
+        user_id: '123',
+        session_id: 'verylongsesssionid',
+        remote_address: '127.0.0.1',
+        properties: {
+          env: '3.141'
+        }
+      }
+      context = Unleash::Context.new(context_params)
+      constraint = Unleash::Constraint.new('env', 'NUM_EQ', '3.141')
+      expect(constraint.matches_context?(context)).to be_truthy
+
+      constraint = Unleash::Constraint.new('env', 'NUM_EQ', '2.718')
+      expect(constraint.matches_context?(context)).to be_falsey
+    end
+
+    it 'matches based on property NUM_LT value' do
+      context_params = {
+        user_id: '123',
+        session_id: 'verylongsesssionid',
+        remote_address: '127.0.0.1',
+        properties: {
+          env: '3.141'
+        }
+      }
+      context = Unleash::Context.new(context_params)
+
+      constraint = Unleash::Constraint.new('env', 'NUM_LT', '2.718')
+      expect(constraint.matches_context?(context)).to be_falsey
+
+      constraint = Unleash::Constraint.new('env', 'NUM_LT', '3.141')
+      expect(constraint.matches_context?(context)).to be_falsey
+
+      constraint = Unleash::Constraint.new('env', 'NUM_LT', '6.282')
+      expect(constraint.matches_context?(context)).to be_truthy
+    end
+
+    it 'matches based on property NUM_LTE value' do
+      context_params = {
+        user_id: '123',
+        session_id: 'verylongsesssionid',
+        remote_address: '127.0.0.1',
+        properties: {
+          env: '3.141'
+        }
+      }
+      context = Unleash::Context.new(context_params)
+
+      constraint = Unleash::Constraint.new('env', 'NUM_LTE', '2.718')
+      expect(constraint.matches_context?(context)).to be_falsey
+
+      constraint = Unleash::Constraint.new('env', 'NUM_LTE', '3.141')
+      expect(constraint.matches_context?(context)).to be_truthy
+
+      constraint = Unleash::Constraint.new('env', 'NUM_LTE', '6.282')
+      expect(constraint.matches_context?(context)).to be_truthy
+    end
+
+    it 'matches based on property NUM_GT value' do
+      context_params = {
+        user_id: '123',
+        session_id: 'verylongsesssionid',
+        remote_address: '127.0.0.1',
+        properties: {
+          env: '3.141'
+        }
+      }
+      context = Unleash::Context.new(context_params)
+
+      constraint = Unleash::Constraint.new('env', 'NUM_GT', '2.718')
```

Comment:
maybe it could be nice to have extra scenarios when using higher precision (say 4, or 6 decimal cases), so we also confirm how float rounding is done as expected?

Reply:
So I've added a test case checking if (0.1 + 0,2).to_s is numerically equal to 0.3, this should be one of the edge cases where the computation doesn't match 0,3 exactly but should still resolve correctly. I don't want to add too many more because it feels like we're checking Ruby's floating point spec and we're checking against Float::EPSILON in the lambda. But I'm open to adding if folks feel strongly about it